To-do list/Models/Helpers/APIHelper.swift

```diff
@@ -0,0 +1,116 @@
+
+//  APICaller.swift
+//  To-do list
+//
+//  Created by Kyrylo Tokar on 2022-10-17.
+//
+
+import Foundation
+
+class APIHelper {
+    
+    enum ErrorAPI: Error {
+        case badRequest(String)
+        case unauthorized(String)
+        case others(String)
+    }
+    
+    private let baseURL = URL(string: "https://education.octodev.net/api/v1")
+    static var token = ""
+    
+    private var session: URLSession
+    init() {
+        let configuration = URLSessionConfiguration.default
+        session = URLSession(configuration: configuration,
+                             delegate: nil,
+                             delegateQueue: OperationQueue.main)
+    }
+    
+    func createPostRequest<T:Codable>(query: String,
+                                      params: [String: Any],
+                                      isUsedToken: Bool = true,
+                                      completion: @escaping (Result <T, ErrorAPI>) -> Void) {
+        guard let request = configUrlRequest(query: query,
+                                             bodyParams: params,
+                                             httpMethod: "POST",
+                                             isUsedToken: isUsedToken) else { return }
+        fetchRequest(with: request, completion: completion)
+    }
+    
+    func createGetRequest<T:Codable>(query: String,
+                                     completion: @escaping (Result <T, ErrorAPI>) -> Void) {
+        guard let request = configUrlRequest(query: query) else { return }
+        fetchRequest(with: request, completion: completion)
+    }
+    
+    private func configUrlRequest(query: String,
+                                  bodyParams: [String: Any]? = nil,
+                                  httpMethod: String = "GET",
+                                  isUsedToken: Bool = true) -> URLRequest? {
+        guard let url = baseURL?.appendingPathComponent(query) else { return nil }
+        var request = URLRequest(url: url)
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        if isUsedToken {
```

Comment:
instead of checking `isUsedToken`, check if `APIHelper.token != ""` and then use it

Reply:
Or you can create a computed property var isUsedToken: Bool { return !self.token.isEmpty }